compiler/Core/CompilerState.cc

```diff
@@ -114,12 +114,4 @@ void CompilerState::runCheapOptimizations(llvm::Function *func) {
     pm.run(*func);
 }
 
-void failCompilation(const core::GlobalState &gs, const core::Loc &loc, ConstExprStr msg) {
-    if (auto e = gs.beginError(loc, core::errors::Compiler::Unanalyzable)) {
-        e.setHeader(msg);
-    }
-
-    throw AbortCompilation(msg.str);
-}
-
```

Comment:
Moved to its own file, because this needs `gs.beginError` (not enough to just forward declare GlobalState) but also needs to be a template to support format_string checking (required by `setHeader`—there is no non-const version).